CONTRIBUTING.md

```diff
@@ -10,6 +10,27 @@ These can be added to your pre-commit hooks to automate the checks. Beyond these
 
 Besides this, we do not have any specific contribution guidelines or codes of conduct for now, however most likely these will be fleshed out as Odilia matures more.
 
+## How To Add A Feature
+
+Odilia has a unique architecture which embeds alot of code into types.
+Here is what you need to know if you want to add a new feature:
+
+- [ ] Find the AT-SPI event that corresponds to the desired feature
+    - Take a look at the docs for `common/events/` and try to find the right event.
+    - For example, to create a feature that reads out changed text in an aria-live region, you would want the event `TextChangedEvent`.
+- [ ] Decide if there are any pre-requisites to the feature being triggerwd (think: focused window only, open tab only, etc.)
+- [ ] See if the prerequisite is already defined
+    - Check out the `odilia/src/tower/cache_event.rs` file for types that implement the `Predicate` trait (defined in the `refinement` create).
+- [ ] Decide what, if any, state is required for your feature (caret position, last focused item, etc.)
+    - For example: to know if the current window is focused, the current window must be stored in the state of the screen reader.
+- [ ] Check to see if we already have your state info as a type; you can check this at `odilia/src/state.rs`.
+    - If not, make a [newtype](https://doc.rust-lang.org/book/ch19-04-advanced-types.html#using-the-newtype-pattern-for-type-safety-and-abstraction), add it to the `State` struct, then implement the `FromAsyncState` trait for the type (so it can be extracted by the `FromAsyncState` implementation).
```

Comment:
I wonder if it wouldn't be better, from a document-source-reading perspective, if this URL were a named reference? You'd change the link here to something like,

```markdown
[newtype][newtype-pat]
```

And waay at the bottom of the file, drop in:
```markdown
[newtype-pat]: https://doc.rust-lang.org/book/ch19-04-advanced-types.html#using-the-newtype-pattern-for-type-safety-and-abstraction
```

Reply:
hmm, that's debatable, as most people would read it in rendered form anyway. Definitely could be done though, if @TTWNO thinks it's better that way. If you read the markdown source though, it's definitely better to do it the way you suggest